3.8.0 broken on illumos

As of version 3.8.0, this crate fails to compile on illumos:

```
error[E0308]: mismatched types
  --> /home/af/.cargo/registry/src/index.crates.io-6f17d22bba15001f/os_info-3.8.0/src/illumos/mod.rs:32:14
   |
31 |     match uname("-o") {
   |           ----------- this expression has type `std::option::Option<std::string::String>`
32 |         Some("illumos") => Type::Illumos,
   |              ^^^^^^^^^ expected `String`, found `&str`
```

--- a/os_info/src/illumos/mod.rs
+++ b/os_info/src/illumos/mod.rs
@@ -28,9 +28,14 @@ fn get_version() -> Option<String> {
 }
 
 fn get_os() -> Type {
-    match uname("-o") {
-        Some("illumos") => Type::Illumos,
-        None => Type::Unknown,
+    let os = match uname("-o") {
+        Some(o) => o,
+        None => return Type::Unknown,
+    };
+
+    match os.as_str() {
+        "illumos" => Type::Illumos,
+        _ => Type::Unknown
     }
 }
 

